docs/rules/no-parameter-properties.md

```diff
@@ -0,0 +1,342 @@
+}
+
+class Foo {
+    constructor(public name: string) {}
+}
+
+class Foo {
+    constructor(private readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected readonly name: string) {}
+}
+
+class Foo {
+    constructor(public readonly name: string) {}
+}
+```
+
+Examples of **correct** code for this rule with no options at all:
+```ts
+class Foo {
+    constructor(name: string) {}
+}
+```
+
+### readonly
+Examples of **incorrect** code for the `{ "allows": ["readonly"] }` options:
+```ts
+class Foo {
+    constructor(private name: string) {}
+}
+
+class Foo {
+    constructor(protected name: string) {}
+}
+
+class Foo {
+    constructor(public name: string) {}
+}
+
+class Foo {
+    constructor(private readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected readonly name: string) {}
+}
+
+class Foo {
+    constructor(public readonly name: string) {}
+}
+```
+
+Examples of **correct** code for the `{ "allows": ["readonly"] }` options:
+```ts
+class Foo {
+    constructor(name: string) {}
+}
+
+class Foo {
+    constructor(readonly name: string) {}
+}
+```
+
+### private
+Examples of **incorrect** code for the `{ "allows": ["private"] }` options:
+```ts
+class Foo {
+    constructor(readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected name: string) {}
+}
+
+class Foo {
+    constructor(public name: string) {}
+}
+
+class Foo {
+    constructor(private readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected readonly name: string) {}
+}
+
+class Foo {
+    constructor(public readonly name: string) {}
+}
+```
+
+Examples of **correct** code for the `{ "allows": ["private"] }` options:
+```ts
+class Foo {
+    constructor(name: string) {}
+}
+
+class Foo {
+    constructor(private name: string) {}
+}
+```
+
+### protected
+Examples of **incorrect** code for the `{ "allows": ["protected"] }` options:
+```ts
+class Foo {
+    constructor(readonly name: string) {}
+}
+
+class Foo {
+    constructor(private name: string) {}
+}
+
+class Foo {
+    constructor(public name: string) {}
+}
+
+class Foo {
+    constructor(private readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected readonly name: string) {}
+}
+
+class Foo {
+    constructor(public readonly name: string) {}
+}
+```
+
+Examples of **correct** code for the `{ "allows": ["protected"] }` options:
+```ts
+class Foo {
+    constructor(name: string) {}
+}
+
+class Foo {
+    constructor(protected name: string) {}
+}
+```
+
+### public
+Examples of **incorrect** code for the `{ "allows": ["public"] }` options:
+```ts
+class Foo {
+    constructor(readonly name: string) {}
+}
+
+class Foo {
+    constructor(private name: string) {}
+}
+
+class Foo {
+    constructor(protected name: string) {}
+}
+
+class Foo {
+    constructor(private readonly name: string) {}
+}
+
+class Foo {
+    constructor(protected readonly name: string) {}
+}
+
+class Foo {
+    constructor(public readonly name: string) {}
```

Comment:
Should this be allowed if the options are `{ "allows": ["public", "readonly"] }`?